probe-rs/src/debug/unit_info.rs

```diff
@@ -68,48 +68,43 @@ impl UnitInfo {
         let mut entries_cursor = self.unit.entries();
         while let Ok(Some((_depth, current))) = entries_cursor.next_dfs() {
             let Some(mut die) = FunctionDie::new(current.clone(), self) else {
+                // We only want to process DIEs that are functions.
                 continue;
             };
 
-            let mut ranges = debug_info.dwarf.die_ranges(&self.unit, current)?;
-
-            while let Ok(Some(range)) = ranges.next() {
-                if !range.contains(address) {
-                    continue;
-                }
-
-                // Check if we are actually in an inlined function
-                die.low_pc = range.begin;
-                die.high_pc = range.end;
+            let mut gimli_ranges = debug_info.dwarf.die_ranges(&self.unit, current)?;
+            while let Ok(Some(gimli_range)) = gimli_ranges.next() {
+                die.ranges.push(gimli_range.begin..gimli_range.end);
+            }
 
-                // Extract the frame_base for this function DIE.
-                let mut functions = vec![die];
+            if !die.range_contains(address) {
+                continue;
+            }
 
-                tracing::debug!(
-                    "Found DIE: name={:?}",
-                    functions[0].function_name(debug_info)
-                );
+            let mut functions = vec![die];
+            tracing::debug!(
+                "Found DIE: name={:?}",
+                functions[0].function_name(debug_info)
+            );
 
-                if find_inlined {
-                    tracing::debug!("Checking for inlined functions");
+            if find_inlined {
+                tracing::debug!("Checking for inlined functions");
 
-                    let inlined_functions =
-                        self.find_inlined_functions(debug_info, address, current.offset())?;
+                let inlined_functions =
+                    self.find_inlined_functions(debug_info, address, current.offset())?;
 
-                    if inlined_functions.is_empty() {
-                        tracing::debug!("No inlined function found!");
-                    } else {
-                        tracing::debug!(
-                            "{} inlined functions for address {}",
-                            inlined_functions.len(),
-                            address
-                        );
-                    }
-
-                    functions.extend(inlined_functions.into_iter());
+                if inlined_functions.is_empty() {
+                    tracing::debug!("No inlined function found!");
```

Comment:
Since you're already touching this code, do we need to special-case a debug log?

Reply:
Hmmm ... good question. Thinking about it,  the fact that the api asks you to explicitly choose to include inlined functions, this will only happen if you want to get them, which makes me think you would want to know if they aren't there.

Reply:
Sure but printing "0 inlined functions for address XY" seems equally good to me

Reply:
True dat!